### PR TITLE
[MA-526] Add new table for SSO

### PR DESCRIPTION
**Pull request by adentong**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: (link Jira ticket)

#changelog [MA-526] Add new table for SSO

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Setup/UpgradeSchema.php [hunk 1]
```diff
@@ -230,4 +227,30 @@ private function updateWebhookLogTable($setup)
...
+    private function setupExternalCustomerEntityTable($setup)
+    {
+        $tableCreated = $setup->getConnection()->isTableExists('external_customer_entity');
+        if ($tableCreated) {
+            return;
+        }
+
+        $table = $setup->getConnection()
+            ->newTable($setup->getTable('external_customer_entity'))
+            ->addColumn(
```
Comment (nickfiacco-bolt) [inline Setup/UpgradeSchema.php:240]: can you make this reference the `customer` table ID? otherwise this is fine.

**Review by nickfiacco-bolt — APPROVED** [review]

File: Setup/UpgradeSchema.php (hunk 1 above)
Comment (adentong) [reply Setup/UpgradeSchema.php:240]: oh this is jsut the primary key column i'll add a new column for customer_id

Comment (nickfiacco-bolt) [reply Setup/UpgradeSchema.php:240]: ok cool

File: Setup/UpgradeSchema.php [hunk 2]
```diff
@@ -230,4 +227,37 @@ private function updateWebhookLogTable($setup)
             ]
         );
     }
+
+    private function setupExternalCustomerEntityTable($setup)
+    {
+        $tableCreated = $setup->getConnection()->isTableExists('external_customer_entity');
```
Comment (vitaliyreznikov) [inline Setup/UpgradeSchema.php:233]: need bolt_ prefix in table name

Comment (adentong) [reply Setup/UpgradeSchema.php]: added

**Review by nickfiacco-bolt — APPROVED** [review]

**Review by vitaliyreznikov — APPROVED** [review]

**Review by danac-gs — APPROVED** [review]

**Pull request merged**